freestyle/shared/src/test/scala/freestyle/free.scala

```diff
@@ -129,6 +129,25 @@ class freeTests extends WordSpec with Matchers {
       "implicitly[SCtors1[SCtors1.Op]]" should compile 
     }
 
+    "allow implicit parameters in algebra members" in {
```

Comment:
Apart from these test cases, specific for type-classes, we should also add the following test cases:  

* A test case with a simpler implicit parameter: 
```Scala
   "@free trait A { def b(x: Int)(implicit y: Int): FS[Unit]" should compile
```

* A test case with only implicit and no non-implicit parameters: 
```Scala
"@free trait A { def b(implicit y: Int): FS[Unit]" should compile
```